### ISSUE TITLE
NoMethodError: undefined method `composite?'

### ISSUE THREAD

**Issue by jufemaiz** [composite-primary-keys/composite_primary_keys#352]
``` Bash
NoMethodError: undefined method `composite?' for #<Thing::Blorg:0x007fceec2aee88>
from /Users/****/.rvm/gems/ruby-2.0.0-p645@Rails4.2_EnergyLink/gems/activemodel-4.2.0/lib/active_model/attribute_methods.rb:433:in `method_missing'
```

So I've got a model that is not ActiveRecord but uses ActiveModel. It's populated from external API calls but also has relationships to the internal rails application.

``` ruby
class Thing::Blorg
  extend ActiveModel::Naming
  extend ActiveModel::Callbacks
  include ActiveModel::Model
  include ActiveModel::Associations
  include ActiveModel::Conversion
  include ActiveModel::Dirty
  include ActiveModel::Validations

  define_attribute_methods :bob_id
  attr_accessor :bob_id
  belongs_to :bob

end
```

It appears that ActiveModel is trying to call "composite?" despite it not being attached to ActiveModel but ActiveRecord? The error occurs when I try to call the relationship:

``` ruby
blorg = Thing::Blorg.find(100)
blorg.bob
```

I can obviously add in:

``` ruby
def composite?
  false
end
```

However I'd obviously like to work out where the underlying issue is and see if it's somethign that needs resolving here as an edge case or if it has to be resolved in all models using ActiveModel?

**Comment by cfis** [comment]
Yeah, I don't know of anything that would cause that.  Hard to tell without a full traceback though.  Could you please provide one?

**Comment by jufemaiz** [comment]
Hi Charlie, will work on recreating today. Cheers

On Sunday, 17 April 2016, Charlie Savage <notifications@github.com> wrote:

> Yeah, I don't know of anything that would cause that. Hard to tell without
> a full traceback though. Could you please provide one?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/composite-primary-keys/composite_primary_keys/issues/352#issuecomment-210921338>
>


-- 
–
Joel Courtney
 m: +61 401 501 625
 e: euphemize@gmail.com
 w: http://euphemize.net
 p: http://flic.kr/jufemaiz/
 t: http://twitter.com/jufemaiz

**Comment by jufemaiz** [comment]
https://github.com/jufemaiz/testing-composite-primary-keys

It's a conflict with ActiveModel::Associations – https://github.com/joker1007/activemodel-associations

:(

**Comment by jufemaiz** [comment]
```Traceback
NoMethodError: undefined method `composite?' for #<ApiModel:0x007fed73c80998>
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activemodel-4.2.6/lib/active_model/attribute_methods.rb:433:in `method_missing'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/composite_primary_keys-8.1.1/lib/composite_primary_keys/associations/singular_association.rb:7:in `get_records_with_cpk_support'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activerecord-4.2.6/lib/active_record/associations/singular_association.rb:57:in `find_target'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activerecord-4.2.6/lib/active_record/associations/association.rb:138:in `load_target'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activerecord-4.2.6/lib/active_record/associations/association.rb:53:in `reload'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activerecord-4.2.6/lib/active_record/associations/singular_association.rb:9:in `reader'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activerecord-4.2.6/lib/active_record/associations/builder/association.rb:115:in `device'
	from (irb):3
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/railties-4.2.6/lib/rails/commands/console.rb:110:in `start'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/railties-4.2.6/lib/rails/commands/console.rb:9:in `start'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/railties-4.2.6/lib/rails/commands.rb:17:in `<top (required)>'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
... 2 levels...
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:268:in `load'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:268:in `block in load'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:268:in `load'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/commands/rails.rb:6:in `call'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/command_wrapper.rb:38:in `call'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application.rb:191:in `block in serve'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application.rb:161:in `fork'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application.rb:161:in `serve'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application.rb:131:in `block in run'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application.rb:125:in `loop'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application.rb:125:in `run'
	from /Users/joelcourtney/.rvm/gems/ruby-2.3.0@TestingCompositePrimaryKeys/gems/spring-1.7.1/lib/spring/application/boot.rb:19:in `<top (required)>'
	from /Users/joelcourtney/.rvm/rubies/ruby-2.3.0/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/joelcourtney/.rvm/rubies/ruby-2.3.0/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from -e:1:in `<main>'2.3.0 :004 > exit
```

**Comment by joker1007** [comment]
I wrote my opinion at my repository. (joker1007/activemodel-associations#9) 
Thanks.

**Comment by cfis** [comment]
Finally had a chance to look into this.  Took quite a while to work it out, but should be fixed in cc80b96.

Thanks for the report!
